src/Utils.cc

```diff
@@ -382,11 +384,26 @@ void copyChildren(ElementPtr _sdf, tinyxml2::XMLElement *_xml,
 
 /////////////////////////////////////////////////
 std::string resolveURI(const std::string &_inputURI,
-    const sdf::ParserConfig &_config, sdf::Errors &_errors)
+    const sdf::ParserConfig &_config, sdf::Errors &_errors,
+    const std::unordered_set<std::string> &_searchPaths)
 {
   std::string resolvedURI = _inputURI;
   if (_config.StoreResolvedURIs())
   {
+    std::string sep("://");
+    if (!_searchPaths.empty() &&
+        _inputURI.find(sep) == std::string::npos &&
+        !std::filesystem::path(_inputURI).is_absolute())
+    {
+      for (const auto &sp : _searchPaths)
+      {
+        std::string fullPath = sdf::filesystem::append(sp, _inputURI);
+        resolvedURI = sdf::findFile(fullPath, true, false);
+        if (!resolvedURI.empty())
+          return resolvedURI;
+      }
+    }
+
     resolvedURI = sdf::findFile(_inputURI, true, true, _config);
```

Comment:
we could add a comment here about the `true, true` arguments: "Search local paths and use registered callbacks"

Reply:
added comment. [83af965](https://github.com/gazebosim/sdformat/pull/1373/commits/83af965d2375ffe3438385e5b8e73a410db6a359)